metis-pattern-analysis/src/main/java/eu/europeana/patternanalysis/ProblemPatternAnalyzer.java

```diff
@@ -53,51 +71,176 @@ public List<ProblemPattern> analyzeRecord(String rdfString) throws Serialization
    */
   public List<ProblemPattern> analyzeRecord(RDF rdf) {
     final List<ProxyType> providerProxies = getProviderProxies(rdf);
-    final List<String> titles = providerProxies.stream().map(EuropeanaType::getChoiceList).flatMap(Collection::stream)
-                                               .filter(Choice::ifTitle).map(Choice::getTitle)
-                                               .map(Title::getString)
-                                               .filter(StringUtils::isNotBlank)
-                                               .map(String::trim)
-                                               .collect(Collectors.toList());
-    final List<String> descriptions = providerProxies.stream().map(EuropeanaType::getChoiceList).flatMap(Collection::stream)
-                                                     .filter(Choice::ifDescription).map(Choice::getDescription)
-                                                     .map(Description::getString)
-                                                     .filter(StringUtils::isNotBlank)
-                                                     .map(String::trim)
-                                                     .collect(Collectors.toList());
+    final List<Choice> choices = providerProxies.stream().map(EuropeanaType::getChoiceList).flatMap(Collection::stream)
+                                                .collect(toList());
+
+    final List<String> titles = getChoicesInStringList(choices, Choice::ifTitle, Choice::getTitle, LiteralType::getString);
+    final List<String> descriptions = getChoicesInStringList(choices, Choice::ifDescription, Choice::getDescription,
+        ResourceOrLiteralType::getString);
+    final List<String> identifiers = getChoicesInStringList(choices, Choice::ifIdentifier, Choice::getIdentifier,
+        LiteralType::getString);
     final String rdfAbout = rdf.getProvidedCHOList().stream().filter(Objects::nonNull).findFirst()
                                .map(ProvidedCHOType::getAbout).orElse(null);
-    return computeProblemPatterns(rdfAbout, titles, descriptions);
+    return computeProblemPatterns(rdfAbout, titles, descriptions, identifiers);
+  }
+
+  private <T> List<String> getChoicesInStringList(List<Choice> choices, Predicate<Choice> choicePredicate,
+      Function<Choice, T> choiceGetter, Function<T, String> getString) {
+    return choices.stream().filter(choicePredicate).map(choiceGetter).map(getString).collect(toList());
   }
 
-  private ArrayList<ProblemPattern> computeProblemPatterns(String rdfAbout, List<String> titles, List<String> descriptions) {
+  private ArrayList<ProblemPattern> computeProblemPatterns(String rdfAbout, List<String> titles, List<String> descriptions,
+      List<String> identifiers) {
     final ArrayList<ProblemPattern> problemPatterns = new ArrayList<>();
 
```

Comment:
Is it feasable to add pattern name description field to ProblemPatternId? After a while we will be struggling to remember what does every pattern. Adding to the enum a short description, so when as new developer hover over the enum we know what pattern is it.

```
public enum ProblemPatternId {
 P1("Systematic use of the same title"), 
 P2("Equal title and description fields"),  
 P3("Near-Identical title and description fields"), 
 P5("Unrecognisable title"),
 P6("Non-meaningful title"), 
 P7("Missing description fields"), 
 P9("Very short description"),
 P12("Extremely long values") 
}
```

Reply:
Hovering on the enum item does not show the description..
What I can do is add javadoc on each value like so
    /**
     * Equal title and description fields
     */
That will show on hovering. Also I would add it to the ProblemPatternDescription perhaps instead.

Reply:
Personal preference: beter if it is hard coded instead of the comment. Let me know if you agree. Is just to remember what the pattern does.

Reply:
As we discussed yes that would be a preferred approach if the actual value was directly used 'as is' in code somewhere.
There is though the message report that actually is used in code which is slightly different.
For now I've added javadoc comments for each value so on "hover" it will show now.
But I will think about this again for the templates on the formaters with the upcoming ticket P1.

Reply:
ok.